services/github/src/queue-lock.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Implements locked queue to allow one routine running at a time
+ */
+class QueuedLock {
```

Comment:
is there a better name?

Reply:
I wonder if something from https://github.com/sindresorhus/promise-fun would serve this purpose, or at least could be used for the map values?

Reply:
I wasn't able to find anything that matched my use-case